src/cim/components/modals/CimConfiguration/CimConfigurationModal.tsx

```diff
@@ -0,0 +1,253 @@
+    }
+    _setImgVolSize(v);
+  };
+
+  const isEdit = !!agentServiceConfig;
+
+  React.useEffect(
+    () => {
+      const doItAsync = async (): Promise<void> => {
+        if (platform === 'AWS') {
+          if (!isEdit || (await isIngressController(getResource))) {
+            setConfigureLoadBalancer(true);
+            setConfigureLoadBalancerInitial(true);
+            return;
+          }
+        }
+
+        setConfigureLoadBalancer(false);
+        setConfigureLoadBalancerInitial(false);
+      };
+
+      void doItAsync();
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [platform],
+  );
+
+  const formProps: CimConfiguratioProps = {
+    dbVolSize,
+    dbVolSizeValidation,
+    setDbVolSize,
+    fsVolSize,
+    fsVolSizeValidation,
+    setFsVolSize,
+    imgVolSize,
+    imgVolSizeValidation,
+    setImgVolSize,
+
+    configureLoadBalancer,
+    configureLoadBalancerInitial,
+    setConfigureLoadBalancer,
+  };
+
+  const onConfigure = () => {
+    const doItAsync = async (): Promise<void> => {
+      setSaving(true);
+      setError(undefined);
+      resetCimConfigProgressAlertSuccessStatus();
+
+      if (
+        await onEnableCIM({
+          t,
+          setError,
+          getResource,
+          listResources,
+          patchResource,
+          createResource,
+          agentServiceConfig,
+
+          platform,
+          dbVolSize,
+          fsVolSize,
+          imgVolSize,
+          configureLoadBalancer,
+        })
+      ) {
+        // successfuly persisted
+        onClose();
+      }
+
+      // keep modal open and show error
+      setConfigureLoadBalancerInitial(configureLoadBalancer);
+      setSaving(false);
+    };
+
+    void doItAsync();
+  };
+
+  const isError = !!error?.title; // this is a communication error only (not the one from agentServiceConfig)
+
+  const isConfigure =
+    !isEdit || !configureLoadBalancerInitial; /* The only possible change for the Edit flow */
+
+  const isConfigureDisabled = !!(
+    isSaving ||
+    dbVolSizeValidation ||
+    fsVolSizeValidation ||
+    imgVolSizeValidation ||
+    (isEdit && configureLoadBalancerInitial === configureLoadBalancer)
+  );
+
+  const isInProgressPeriod = isCIMConfigProgressing({ agentServiceConfig });
+  const isConfiguring = isSaving || isInProgressPeriod;
+
+  let actions: React.ReactNode[];
+  if (isConfigure) {
+    actions = [
+      <Button
+        key="configure"
+        variant={ButtonVariant.primary}
+        isDisabled={isConfigureDisabled}
+        onClick={onConfigure}
+      >
+        {t('ai:Configure')}
+      </Button>,
+      <Button key="cancel" variant={ButtonVariant.link} onClick={onClose}>
+        {t('ai:Cancel')}
+      </Button>,
+    ];
+  } else if (isConfiguring) {
+    actions = [
+      <Button key="configuring" isDisabled={true /* always */} variant={ButtonVariant.primary}>
+        <Spinner size="sm" />
+        &nbsp;
+        {t('ai:Configuring')}
+      </Button>,
```

Comment:
woudn't it look better by using `icon` ?

```suggestion
      <Button key="configuring" isDisabled={true /* always */} variant={ButtonVariant.primary} icon={<Spinner size="sm" />}>
        {t('ai:Configuring')}
      </Button>,
```

Reply:
Thanks for poking me. There is even `isLoading` property on the Button.

Reply:
done